tests/phpunit/tests/html/wpHtmlTagProcessor-bookmark.php

```diff
@@ -0,0 +1,359 @@
+			  Squash and merge
+			</button>
+
+			<button type="button" class="merge-box-button btn-group-rebase rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Rebase and merge
+			</button>
+
+			<button aria-label="Select merge method" disabled="disabled" type="button" data-view-component="true" class="select-menu-button btn BtnGroup-item"></button>
+		</div>
+	</div>
+</div>
+HTML;
+		$expected_output = <<<HTML
+<div selected class="merge-message" checked>
+	<div class="select-menu d-inline-block">
+		<div  class="BtnGroup MixedCaseHTML position-relative" />
+		<div checked class="BtnGroup MixedCaseHTML position-relative">
+			<button type="submit" class="merge-box-button btn-group-merge rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Merge pull request
+			</button>
+
+			<button  class="hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Squash and merge
+			</button>
+
+			<button id="rebase-and-merge"     disabled="">
+			  Rebase and merge
+			</button>
+
+			<button id="last-button"     ></button>
+		</div>
+	</div>
+</div>
+HTML;
+		$p               = new WP_HTML_Tag_Processor( $input );
+		$p->next_tag( 'div' );
+		$p->next_tag( 'div' );
+		$p->next_tag( 'div' );
+		$p->set_bookmark( 'first div' );
+		$p->next_tag( 'button' );
+		$p->set_bookmark( 'first button' );
+		$p->next_tag( 'button' );
+		$p->set_bookmark( 'second button' );
+		$p->next_tag( 'button' );
+		$p->set_bookmark( 'third button' );
+		$p->next_tag( 'button' );
+		$p->set_bookmark( 'fourth button' );
+
+		$p->seek( 'first button' );
+		$p->set_attribute( 'type', 'submit' );
+
+		$this->assertTrue(
+			$p->seek( 'third button' ),
+			'Seek() to the third button failed'
+		);
+		$p->remove_attribute( 'class' );
+		$p->remove_attribute( 'type' );
+		$p->remove_attribute( 'aria-expanded' );
+		$p->set_attribute( 'id', 'rebase-and-merge' );
+		$p->remove_attribute( 'data-details-container' );
+
+		$this->assertTrue(
+			$p->seek( 'first div' ),
+			'Seek() to the first div failed'
+		);
+		$p->set_attribute( 'checked', false );
+
+		$this->assertTrue(
+			$p->seek( 'fourth button' ),
+			'Seek() to the fourth button failed'
+		);
+		$p->set_attribute( 'id', 'last-button' );
+		$p->remove_attribute( 'class' );
+		$p->remove_attribute( 'type' );
+		$p->remove_attribute( 'checked' );
+		$p->remove_attribute( 'aria-label' );
+		$p->remove_attribute( 'disabled' );
+		$p->remove_attribute( 'data-view-component' );
+
+		$this->assertTrue(
+			$p->seek( 'second button' ),
+			'Seek() to the second button failed'
+		);
+		$p->remove_attribute( 'type' );
+		$p->set_attribute( 'class', 'hx_create-pr-button' );
+
+		$this->assertSame(
+			$expected_output,
+			$p->get_updated_html()
+		);
```

Comment:
Requires `$message`.

Reply:
```suggestion
		$this->assertSame(
			$expected_output,
			$p->get_updated_html(),
			"Performing several attribute updates on different tags does not produce the expected HTML snippet."
		);
```

Reply:
reviewed all tests in 5b4ad8fa56 to ensure they have a message

surprisingly, the vast majority of existing Core code is missing these.